### PR TITLE
Move method from Vertex to Graph

### LINKED ISSUE #31: Move method from Vertex
Find a good place to put Vertex.checkIsolation() because it's not confortable right now. 
Consider its intent and give it a good home.
Thanks

### PR DESCRIPTION
**Pull request by rmilfont**
Description: Change the location of checkIsolation method. Put it at
Graph, nested with haveIsolationCores.
Commit:Mod
TaskNumber:#31

### CODE REVIEW

File: src/util/Graph.java [hunk 1]
```diff
@@ -14,22 +14,25 @@ public Graph(int rows, int columns) {
 		dimX = columns;
 		dimY = rows;
 	}
-	
+
 	public boolean haveIsolatedCores() {
```
Comment (jmferreiratech) [inline src/util/Graph.java]: You could rename it to "hasIsolatedVertices".

File: src/util/Graph.java [hunk 2]
```diff
@@ -14,22 +14,25 @@ public Graph(int rows, int columns) {
...
-		
-    	return false;
+		Set<Vertex> reachable = new HashSet<>();
+		checkIsolation(vertex("0.0"), reachable);
+		return !(reachable.size() == vertices.size());
+	}
+
+	private void checkIsolation(Vertex vertex, Set<Vertex> reachable) {
+		for (Edge adj : vertex.adjuncts()){
+			if (reachable.contains(adj.destination()))
```
Comment (jmferreiratech) [inline src/util/Graph.java]: Put this condition outside the for (first thing on this method) checking for "vertex" and returning if it's already present.

File: src/util/Graph.java [hunk 3]
```diff
@@ -14,22 +14,25 @@ public Graph(int rows, int columns) {
...
+				continue;
+			Vertex neigh = adj.destination();
+			reachable.add(neigh);
```
Comment (jmferreiratech) [inline src/util/Graph.java]: Put this add before the for, adding the "vertex".

File: src/util/Graph.java [hunk 4]
```diff
@@ -14,22 +14,25 @@ public Graph(int rows, int columns) {
...
 	public boolean haveIsolatedCores() {
-		ArrayList<Vertex> alc = new ArrayList<Vertex>();
-		//Escolha do 0.0 para ser o core inicial. Garantido a existencia do primeiro nodo em todas as topologias
-		vertex("0.0").checkIsolation(alc);
-		
-		//Se lista de alcancaveis for igual ao total de cores nao existe isolamento
-		if(!(alc.size()==vertices.size())) return true;
-		
-    	return false;
+		Set<Vertex> reachable = new HashSet<>();
```
Comment (jmferreiratech) [inline src/util/Graph.java]: Rename it to reachedVertices

File: src/util/Graph.java [hunk 5]
```diff
@@ -14,22 +14,25 @@ public Graph(int rows, int columns) {
...
+		checkIsolation(vertex("0.0"), reachable);
+		return !(reachable.size() == vertices.size());
+	}
+
+	private void checkIsolation(Vertex vertex, Set<Vertex> reachable) {
```
Comment (jmferreiratech) [inline src/util/Graph.java]: This method name makes no sense.

**Review by jmferreiratech — CHANGES_REQUESTED** [review]

File: src/util/Graph.java (hunk 1 above)
Comment (rmilfont) [reply src/util/Graph.java]: Done.

File: src/util/Graph.java (hunk 4 above)
Comment (rmilfont) [reply src/util/Graph.java]: Done.

File: src/util/Graph.java (hunk 3 above)
Comment (rmilfont) [reply src/util/Graph.java]: Done.

File: src/util/Graph.java (hunk 2 above)
Comment (rmilfont) [reply src/util/Graph.java]: Done.

File: src/util/Graph.java (hunk 5 above)
Comment (rmilfont) [reply src/util/Graph.java]: Done.

**Review by jmferreiratech — APPROVED** [review]

**Pull request merged**